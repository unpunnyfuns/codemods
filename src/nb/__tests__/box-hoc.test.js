import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('box HOC migration', () => {
  it('migrates HOC-wrapped Box (Animated.createAnimatedComponent)', () => {
    const output = testTransform('nb/box', 'nb/box-hoc', 'tsx', {
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
