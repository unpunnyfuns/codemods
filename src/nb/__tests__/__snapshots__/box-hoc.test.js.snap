// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`box HOC migration > migrates HOC-wrapped Box (Animated.createAnimatedComponent) 1`] = `
"import Animated from 'react-native-reanimated'

import { View, StyleSheet } from 'react-native';
import { space } from '@design-tokens';

const AnimatedBox = Animated.createAnimatedComponent(View)

export function Example() {
  return (
    <>
      <View style={styles.box0}>Direct Box</View>
      <AnimatedBox style={styles.box1}>
        Animated Box
      </AnimatedBox>
    </>
  );
}

const styles = StyleSheet.create({
  box0: {
    marginBottom: space.lg
  },

  box1: {
    marginTop: space.md,
    padding: space.sm
  }
});
"
`;
